common/force_send_fields.go

```diff
@@ -0,0 +1,71 @@
+package common
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"golang.org/x/exp/slices"
+)
+
+// SetForceSendFields adds any fields specified in the `fields` parameter to the ForceSendFields field of the
+// request structure if they are present in the resource state. The provided fields must match the JSON tag
+// for some field in the request structure. This ensures that fields explicitly set to the zero value of its
+// type (e.g. `0` for an `int`) will be serialized and sent to the platform.
+//
+// This function requires that the request structure has a `ForceSendFields` field of type `[]string`. If not,
+// it panics with an appropriate error message.
+func SetForceSendFields(req any, d attributeGetter, fields []string) {
+	rv := reflect.ValueOf(req)
+	if rv.Kind() != reflect.Ptr {
+		panic("request argument to setForceSendFields must be a pointer")
```

Comment:
I think we should panic here because this is a programmer error/error in the provider. Note that `common.Resource` catches panics and converts them to errors.